lib/src/rules/always_specify_types.dart

```diff
@@ -163,8 +204,57 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   @override
   void visitVariableDeclarationList(VariableDeclarationList list) {
-    if (list.type == null) {
-      rule.reportLintForToken(list.keyword);
+    var keyword = list.keyword;
+    if (list.type == null && keyword != null) {
+      List<String>? types;
+      var parent = list.parent;
+      if (parent is TopLevelVariableDeclaration) {
+        types = _getTypes(parent.variables);
+      } else if (parent is ForPartsWithDeclarations) {
+        types = _getTypes(parent.variables);
+      } else if (parent is FieldDeclaration) {
+        types = _getTypes(parent.fields);
+      } else if (parent is VariableDeclarationStatement) {
+        types = _getTypes(parent.variables);
+      }
+
+      if (types == null) return;
+
+      var singleType = types.toSet().length == 1;
+
+      List<Object> arguments;
+      ErrorCode? errorCode;
+      if (types.isEmpty) {
+        arguments = [];
+      } else if (keyword.type == Keyword.VAR) {
+        if (singleType) {
+          arguments = [keyword, types.first];
+          errorCode = keywordCouldBeTypeCode;
+        } else {
+          arguments = [keyword];
+          errorCode = keywordCouldBeSplitToTypesCode;
+        }
+      } else {
+        if (singleType) {
+          arguments = [types.first];
+          errorCode = specifyTypeCode;
+        } else {
+          arguments = [];
+        }
+      }
+      rule.reportLintForToken(keyword,
+          arguments: arguments, errorCode: errorCode);
+    }
+  }
+
+  List<String> _getTypes(VariableDeclarationList list) {
+    var types = <String>[];
```

Comment:
The fact that `types` is a list makes me suspect that we won't catch cases where all of the types are the same, such as
```
var a = 0, b = 1;
```
This should produce a diagnostic that suggests the type `int`.

I suspect that you need to use a set here.

Reply:
The List was converted to `Set` in the caller function, now `Set` is directly used.